### ISSUE TITLE
Animations + Combat System

### ISSUE THREAD

**Issue by CarrotKutay** [CarrotKutay/Colosseum#5]
#### Basic Movement
- [x] Walking
  - [x] Forward (+ Running)
  - [x] Backward
  - [x] Left
  - [x] Right

-> put basic movement inside a blend tree for now
-> extension possible with either different layer masks or adding IK-constraints 

#### Dodge Roll Animation
- [ ] We need a dodge mechanic + animation

#### Attack animation
-  [ ] Attack
  - [ ] 1st Attack
  - [ ] 2nd Attack

-> various skill animation needed, once skills are included into combat

**Comment by CarrotKutay** [comment]
There is a current error with instantiating entities from game objects
![Capture_error_BoneIndexOffset](https://user-images.githubusercontent.com/43205636/81607584-34b7cf80-93d5-11ea-8230-5b95c149463e.PNG)

error/bug seems to be known, but not solved as of yet:
https://forum.unity.com/threads/vertex-shader-based-skinned-character-animation.692077/

Found a workaround to suppress the error, but instantiating the entity might be incomplete due to it

**Comment by CarrotKutay** [comment]
@ismaelKutay  probably better to wait on implementing DOTS with animations as there seems to be no easy get-to solution. The only solution I have found so far is to have a custom shader, sperating the frames of the animation and telling each entity which frame to play at what moment. This might be possible for simple animations, but as we are looking for stuff to cope with animation blending between different animations I'd say we put a stop to DOTS for animated objects

https://www.youtube.com/watch?v=tvi44I_SK3w 

or it might be possible to do a hybrid between 'regular' and DOTS, where we perform animations on the entity with the 'normal' MonoBehaviour' workflow

**Comment by CarrotKutay** [comment]
will commence to work with game objects only for now, when working on player movement and animation. Arguably there will be a limited number of players and not too much data handled by them, which might be a bit too much overhead work for making it work in DOTS for now

**Comment by CarrotKutay** [comment]
will put this issue on hold and see if #10  fruits in reasonable results, which might lead to possibilities to pick up DOTS again
